handlers/handlers.go

```diff
@@ -577,33 +577,53 @@ func legacyLanding(w http.ResponseWriter, req *http.Request, zc ZebedeeClient, d
 		log.Warn(ctx, "unable to get homepage content", log.FormatErrors([]error{err}), log.Data{"homepage_content": err})
 	}
 
+	var wg1 sync.WaitGroup
+
 	var datasets []zebedee.Dataset
+	var mutex1 = &sync.Mutex{}
 	for _, v := range dlp.Datasets {
-		dataset, err := zc.GetDataset(ctx, userAccessToken, collectionID, lang, v.URI)
-		if err != nil {
-			setStatusCode(req, w, errors.Wrap(err, "zebedee client legacy dataset returned an error"))
-			return
-		}
+		wg1.Add(1)
+
+		go func() {
+			defer wg1.Done()
+
+			dataset, err := zc.GetDataset(ctx, userAccessToken, collectionID, lang, v.URI)
```

Comment:
`err` is created here, so will not be reflected outside the goroutine (line 611)